### ISSUE TITLE
build failed on mac m1 platform: /Frameworks/VLCKit.framework/Versions/A/VLCKit' (no such file)

### ISSUE THREAD

**Issue by itning** [alexmercerind/dart_vlc#260]
**Describe the bug**
can not build app on mac m1 platform

/Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/build/macos/Build/Products/Debug/test_vlc.app/Contents/Frameworks/dart_vlc.framework/Versions/A/../Frameworks/VLCKit.framework/Versions/A/VLCKit' (no such file)

**Minimal reproducible code**
```dart
import 'package:dart_vlc/dart_vlc.dart';
import 'package:flutter/material.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  DartVLC.initialize();
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(),
        body: Container(),
      ),
    );
  }
}

```

**Flutter logs**
```
Launching lib/main.dart on macOS in debug mode...
Running pod install...
Building macOS application...
2022-04-23 13:58:02.900 xcodebuild[31142:2348294] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-04-23 13:58:02.900 xcodebuild[31142:2348294] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id:BAF653FB-24AD-5EEB-8A2C-CE5CFA7E473D }
{ platform:macOS, name:Any Mac }
-- The C compiler identification is AppleClang 13.1.6.13160021
-- The CXX compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Pods/dartvlc_core
/Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Pods
[  0%] Built target LIBVLC_EXTRACT
[ 33%] Building CXX object CMakeFiles/dart_vlc_core.dir/main.cc.o
[ 66%] Building CXX object CMakeFiles/dart_vlc_core.dir/api/api.cc.o
/Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Flutter/ephemeral/.symlinks/plugins/dart_vlc/dartvlc/api/api.cc:41:48: warning: cannot delete expression with pointer-to-'void' type 'void *' [-Wdelete-incomplete]
static void DestroyObject(void*, void* peer) { delete peer; }
                                               ^      ~~~~
1 warning generated.
[100%] Linking CXX static library libdart_vlc_core.a
[100%] Built target dart_vlc_core
Install the project...
-- Install configuration: ""
-- Installing: /Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Pods/../Flutter/ephemeral/.symlinks/plugins/dart_vlc/macos/deps/lib/libdart_vlc_core.a
install_name_tool: warning: changes being made to the file will invalidate the code signature in: /Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Pods/VLCKit/VLCKit.xcframework/macos-arm64_x86_64/VLCKit.framework/VLCKit (for architecture x86_64)
install_name_tool: warning: changes being made to the file will invalidate the code signature in: /Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/macos/Pods/VLCKit/VLCKit.xcframework/macos-arm64_x86_64/VLCKit.framework/VLCKit (for architecture arm64)
note: Using new build system
note: Planning
note: Build preparation complete
note: Building targets in dependency order
dyld[31903]: Library not loaded: @loader_path/../Frameworks/VLCKit.framework/Versions/A/VLCKit
  Referenced from: /Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/build/macos/Build/Products/Debug/test_vlc.app/Contents/Frameworks/dart_vlc.framework/Versions/A/dart_vlc
  Reason: tried: '/Users/ning.wang/Documents/SourceCode/StudioProjects/test_vlc/build/macos/Build/Products/Debug/test_vlc.app/Contents/Frameworks/dart_vlc.framework/Versions/A/../Frameworks/VLCKit.framework/Versions/A/VLCKit' (no such file), '/Library/Frameworks/VLCKit.framework/Versions/A/VLCKit' (no such file), '/System/Library/Frameworks/VLCKit.framework/Versions/A/VLCKit' (no such file)
Error waiting for a debug connection: The log reader stopped unexpectedly, or never started.
Error launching application on macOS.
```
```
[✓] Flutter (Channel stable, 2.10.5, on macOS 12.3.1 21E258 darwin-arm, locale zh-Hans-CN)
    • Flutter version 2.10.5 at /Users/ning.wang/Documents/SourceCode/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5464c5bac7 (5 days ago), 2022-04-18 09:55:37 -0700
    • Engine revision 57d3bac3dd
    • Dart version 2.16.2
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/ning.wang/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Users/ning.wang/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/211.7628.21.2111.8309675/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Users/ning.wang/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/211.7628.21.2111.8309675/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at /Users/ning.wang/Applications/JetBrains Toolbox/IntelliJ IDEA Ultimate.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Ultimate Edition (version 2022.1)
    • IntelliJ at /Users/ning.wang/Library/Application Support/JetBrains/Toolbox/apps/IDEA-U/ch-0/221.5080.210/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.66.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 100.0.4896.127

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
Process finished with exit code 0
```

**Operating system:**
- Platform: mac os
- OS version: Apple M1 Pro 12.3.1

**Other details:**

- [ ] I confirm this is not a bug in the VLC app & only `dart_vlc`.
- [ ] I have donated / sponsored `dart_vlc`.

**Comment by mkgor** [comment]
got same error, please help

**Comment by milkyway044** [comment]
+1

**Comment by alexmercerind** [comment]
#242

**Comment by fzyzcjy** [comment]
Same error here. https://github.com/fzyzcjy/flutter_convenient_test/issues/186

**Comment by GroovinChip** [comment]
Closed per #303
